student/src/main/java/tds/student/services/remote/RemoteResponseService.java

```diff
@@ -137,54 +149,74 @@ public void markItemForReview(final OpportunityInstance opportunityInstance, fin
     DateTimeFormatter dateFormatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSSSSS");
 
-    for (ExamPage page : examPages) {
-      for (ExamItem item : page.getExamItems()) {
-        OpportunityItem opportunityItem = new OpportunityItem();
-        opportunityItem.setBankKey(item.getAssessmentItemBankKey());
-        opportunityItem.setItemKey(item.getAssessmentItemKey());
-        opportunityItem.setPosition(item.getPosition());
-        opportunityItem.setPage(page.getPagePosition());
-        opportunityItem.setGroupID(page.getItemGroupKey());
-        opportunityItem.setSegment(page.getSegmentPosition());
-        opportunityItem.setSegmentID(page.getSegmentId());
-        opportunityItem.setGroupItemsRequired(page.isGroupItemsRequired() ? -1 : 0);
-        opportunityItem.setIsRequired(item.isRequired());
-        opportunityItem.setItemFile(item.getItemFilePath());
-        opportunityItem.setStimulusFile(item.getStimulusFilePath().isPresent() ? item.getStimulusFilePath().get() : null);
-        opportunityItem.setDateCreated(item.getCreatedAt().toString(dateFormatter));
-
-        // This value is always set to 'format' in the t_getopportunityitems stored procedure, then converted to
-        // upper-case in ResponseRepository#readOpportunityItems() (@ line 295).
-        opportunityItem.setFormat("FORMAT");
-
-        // OpportunityItem#isVisible (mapped to an OpportunityItem in ResponseRepository#readOpportunityItems @ line
-        // 297) never seems to be used (which is why it was not ported to the exam database), so has been omitted from
-        // this mapping.
-
-        if (item.getResponse().isPresent()) {
-          ExamItemResponse itemResponse = item.getResponse().get();
-          opportunityItem.setSequence(itemResponse.getSequence());
-          opportunityItem.setMarkForReview(itemResponse.isMarkedForReview());
-          opportunityItem.setValue(itemResponse.getResponse());
-          opportunityItem.setIsValid(itemResponse.isValid());
-          opportunityItem.setIsSelected(itemResponse.isSelected());
-        } else {
-          opportunityItem.setValue(null);
+    for (ExamSegmentWrapper examSegmentWrapper : examSegmentWrappers) {
+      ExamSegment examSegment = examSegmentWrapper.getExamSegment();
+      for (ExamPageWrapper pageWrapper : examSegmentWrapper.getExamPages()) {
+        ExamPage page = pageWrapper.getExamPage();
+        for (ExamItem item : pageWrapper.getExamItems()) {
+          OpportunityItem opportunityItem = new OpportunityItem();
+          opportunityItem.setBankKey(item.getAssessmentItemBankKey());
+          opportunityItem.setItemKey(item.getAssessmentItemKey());
+          opportunityItem.setPosition(item.getPosition());
+          opportunityItem.setPage(page.getPagePosition());
+          opportunityItem.setGroupID(page.getItemGroupKey());
+          opportunityItem.setSegment(examSegment.getSegmentPosition());
```

Comment:
ExamPage doesn't have a segmentPosition or a segmentId.  Those are on the ExamSegment.  This was a problem that has been fixed on the exam service with this change leveraging it.